R/dplyr.R

```diff
@@ -157,8 +157,21 @@ slice.dm <- function(.data, ...) {
 }
 
 #' @export
-slice.zoomed_dm <- function(.data, ...) {
-  replace_zoomed_tbl(.data, slice(get_zoomed_tbl(.data), ...))
+slice.zoomed_dm <- function(.data, idx) {
+  if (is_missing(idx)) return(.data)
+
+  if (!inherits(idx, "integer")) {
```

Comment:
I forgot to mention that we can use `vctrs::vec_as_index()` to create a positive index vectors from all inputs.

Reply:
would that mean that different from `dplyr::slice()` we would accept logical vectors too(?), or should we actively check against that?

Reply:
I would opt for not accepting logical vectors, since people might get into habits that don't work with `dplyr::slice()`

Reply:
I kept the `class()`-checks, for reasons mentioned in the comment in line 162